rllib/agents/sac/sac_tf_policy.py

```diff
@@ -310,22 +309,48 @@ def gradients(policy, optimizer, loss):
             var_list=[policy.model.log_alpha],
             clip_val=policy.config["grad_clip"])
     else:
-        actor_grads_and_vars = policy._actor_optimizer.compute_gradients(
-            policy.actor_loss, var_list=policy.model.policy_variables())
-        if policy.config["twin_q"]:
-            q_variables = policy.model.q_variables()
-            half_cutoff = len(q_variables) // 2
-            base_q_optimizer, twin_q_optimizer = policy._critic_optimizer
-            critic_grads_and_vars = base_q_optimizer.compute_gradients(
-                policy.critic_loss[0], var_list=q_variables[:half_cutoff]
-            ) + twin_q_optimizer.compute_gradients(
-                policy.critic_loss[1], var_list=q_variables[half_cutoff:])
+        # Eager: Use GradientTape.
+        if tf.executing_eagerly():
```

Comment:
This is pretty ugly, is there any way to not copy paste the code for TFE / non eager?

Reply:
^^ Refer to my comment above; an if statement for function call (optimizer.tape.gradient versus optimizer.get_gradients)

Reply:
It's more problematic, I think:
We have three optimizers for the graph-mode case, but only one (faked by the caller of gradients_fn and carrying the tape object) for the eager case.
You could create a dict/tuple (each optimizer holding grad-fn plus vars), but I think that's even uglier :( .

Reply:
I'm not too familiar with tfe, but is it possible to create 3 optimizers for SAC in tfe? Will it affect the algorithm?